modules/tinymce/src/core/main/ts/textpatterns/core/BlockPatternUtils.ts

```diff
@@ -0,0 +1,91 @@
+import { Arr, Obj, Optional, Type, Unicode } from '@ephox/katamari';
+
+import * as TextSearch from '../../alien/TextSearch';
+import DOMUtils from '../../api/dom/DOMUtils';
+import Editor from '../../api/Editor';
+import Formatter from '../../api/Formatter';
+import * as Options from '../../api/Options';
+import Tools from '../../api/util/Tools';
+import { generatePathRange, resolvePathRange } from '../utils/PathRange';
+import * as Utils from '../utils/Utils';
+import { BlockPattern, BlockPatternMatch, Pattern, PatternSet } from './PatternTypes';
+
+type StripPattern = (dom: DOMUtils, block: Node, pattern: BlockPattern) => void;
+type FindPattern = <P extends Pattern>(patterns: P[], text: string) => Optional<P>;
+
+const stripPattern = (dom: DOMUtils, block: Node, pattern: BlockPattern): Optional<Text> => {
+  // The pattern could be across fragmented text nodes, so we need to find the end
+  // of the pattern and then remove all elements between the start/end range
+  const firstTextNode = TextSearch.textAfter(block, 0, block);
+  return firstTextNode.map((spot) => {
```

Comment:
```suggestion
  TextSearch.textAfter(block, 0, block).map((spot) => {
```

Reply:
Since if the variable is never used except at this spot we might as well get rid of it.

Reply:
done